dask/array/reblock.py

```diff
@@ -0,0 +1,159 @@
+			try:
+				n = next(nw)
+				if not ret or ret[-1]:
+					ret.append([])
+			except StopIteration:
+				break
+			sum_new += n
+			if sum_new - n < sum_old and sum_new >= sum_old:
+				start = o -leftover
+				endd = o
+				adding = (cnt, slice(start, endd))
+				ret[-1].append(adding)
+		leftover = 0
+		while sum_old < sum_new:
+			try:
+				cnt, o = next(od)
+			except StopIteration:
+				break
+			sum_old += o
+			if sum_old > sum_new and sum_old - o < sum_new:
+				leftover =  (sum_old - sum_new)
+				endd = o - leftover
+				start = 0
+			elif sum_old <= sum_new:
+				start = leftover
+				endd = o
+			elif sum_old < sum_new:
+				start = 0
+				endd = o
+			else:
+				continue
+			adding = (cnt, slice(start, endd))
+			ret[-1].append(adding)
+		if sum_old == summ and sum_new == summ:
+			break
+	return tuple(map(tuple, ret))
+
+
+def intersect_blockdims(old_blockdims=None, 
+						new_blockdims=None,
+						shape=None,
+						old_blockshape=None,
+						new_blockshape=None):
+	""" 
+	Make dask.array slices as intersection of old and new blockdims.
+
+	>>> intersect_blockdims(((4,4),(2,)), ((8,), (1,1)))
+		(((((0, slice(0, 4, None)), (1, slice(0, 4, None))),),
+		  (((0, slice(0, 1, None)),), ((0, slice(1, 2, None)),))),
+		 ((((0, slice(0, 4, None)), (0, slice(0, 1, None))),
+		   ((1, slice(0, 4, None)), (0, slice(0, 1, None)))),
+		  (((0, slice(0, 4, None)), (0, slice(1, 2, None))),
+		   ((1, slice(0, 4, None)), (0, slice(1, 2, None))))))
+	
+	Parameters:
+
+	old_blockdims : iterable of tuples
+        block sizes along each dimension (convert from old_blockdims)
+    new_blockdims: iterable of tuples
+    	block sizes along each dimension (converts to new_blockdims) 
+    shape : tuple of ints
+        Shape of the entire array (not needed if using blockdims)
+    old_blockshape: size of each old block as tuple 
+    	(converts from this old_blockshape)
+    new_blockshape: size of each new block as tuple 
+    	(converts to this old_blockshape)
+
+    Note: shape is only required when using old_blockshape or new_blockshape.
+	"""
+	if not old_blockdims:
+		old_blockdims = blockdims_from_blockshape(shape, old_blockshape)
+	if not new_blockdims:
+		new_blockdims = blockdims_from_blockshape(shape, new_blockshape)	
+	zipped = zip(old_blockdims,new_blockdims)
+	old_to_new =  tuple(intersect_blockdims_1d(o, n) for o,n in zipped)
+	cross = tuple(product(*old_to_new))
+	cross = tuple(chain(tuple(product(*cr)) for cr in cross))
+	return cross
+
+
+def reblock(x, blockdims=None, blockshape=None ):
+	"""
+	Convert blocks in dask array x for new blockdims.
+
+	reblock(x, blockdims=None, blockshape=None )
+	
+	>>> import dask.array as da 
+	>>> old_blockdims = ((2,3,2),)*4
+    >>> new = ((2,4,1), (4, 2, 1), (4, 3), (7,))
+    >>> a = np.random.uniform(0, 1, 7**4).reshape((7,) * 4)
+    >>> x = da.from_array(a, blockdims=old)
+    >>> x.blockdims
+    ((2,4,1), (4, 2, 1), (4, 3), (7,))
+
+    Parameters:
+
+    x:   dask array
```

Comment:
Format for parameters is 

```
name: type 
    short description
```

See numpydoc https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt